src/jit/lsra.cpp

```diff
@@ -6836,7 +6943,23 @@ void LinearScan::freeRegister(RegRecord* physRegRecord)
             // we wouldn't unnecessarily link separate live ranges to the same register.
             if (nextRefPosition == nullptr || RefTypeIsDef(nextRefPosition->refType))
             {
-                unassignPhysReg(physRegRecord, nullptr);
+#ifdef _TARGET_ARM_
+                if (assignedInterval->registerType == TYP_DOUBLE && !genIsValidDoubleReg(physRegRecord->regNum))
```

Comment:
I think that, when a register that is assigned to a `TYP_DOUBLE` `Interval` is freed, only the valid (lower) register should be added to the free set. Then this can be an assert, i.e.:
```
assert((assignedInterval->registerType != TYP_DOUBLE) || genIsValidDoubleReg(physRegRecord->regNum));
```